src/convert_hl7v2_fhir/controllers/er7/er7_extractor.py

```diff
@@ -0,0 +1,115 @@
+from datetime import datetime, timezone
+from typing import List
+import hl7apy
+from hl7apy.core import Message
+
+from convert_hl7v2_fhir.controllers.er7.exceptions import (
+    MissingNHSNumberError,
+    InvalidNHSNumberError,
+    MissingPointOfCareError,
+    MissingFacilityError,
+    MissingPatientClassError,
+    MissingAdmissionTypeError,
+    MissingTimeOfAdmissionError,
+    MissingFamilyNameError,
+    MissingGivenNameError,
+)
+from convert_hl7v2_fhir.controllers.utils import is_nhs_number_valid
+
+
+class ER7Extractor:
+    def __init__(self, er7_message: Message):
+        self.er7_message = er7_message
+
+    def nhs_number(self) -> str:
+        """There are two potential fields where NHS number can be:
+        1. PID.2 - Patient Id
+        2. PID.3 - Patient Identifier List"""
+
+        nhs_numbers = [
+            composite_id.id.value
+            for composite_id in self.er7_message.pid.patient_id_internal_id
+            if "NHSNMBR" in composite_id.value
+        ]
+        if "NHSNMBR" in self.er7_message.pid.patient_id_external_id.value:
+            nhs_numbers.append(self.er7_message.pid.patient_id_external_id.id.value)
```

Comment:
I can comment these out with no tests failing.  We probably want an example that shows this bit too.

Reply:
@henrywrightPSC this is a bit icky, because according to docs datatype here is just a string. Yet we treat it as [CX - Extended Composite Id With Check Digit](https://hl7-definition.caristix.com/v2/HL7v2.8/DataTypes/CX)

Perhaps we should remove the logic of extracting NHS number from this field?

Reply:
I think it's just been moved to be a ST datatype because they've withdrawn it (i.e. to say any string can be here and we don't care) but [here it says we can still have a CX for v2.4](https://hl7-definition.caristix.com/v2/HL7v2.4/Segments/PID).

It's a bit chicken and egg because it depends on the pilot partner we go with - certainly Homerton still had PID.2 as a CX type.